manifests/centosimage.pp

```diff
@@ -29,18 +30,32 @@
     before  => Docker::Image['centosagent']
   }
 
+  # Mount the gem cache directory because docker doesn't follow symlinks
+  # mount --bind /var/cache/rubygems rubygems
+  file {'/var/docker/centosagent/rubygems':
+    ensure => directory,
+    before => Mount['/var/docker/centosagent/rubygems'],
+  }
+
+  mount {'/var/docker/centosagent/rubygems':
+    ensure  => mounted,
+    atboot  => true,
+    device  => '/var/cache/rubygems',
+    fstype  => 'xfs',
+    options => 'bind',
+    before  => Docker::Image['centosagent']
+  }
+
   file { '/var/docker/centosagent/Dockerfile':
     ensure  => present,
     content => epp('puppetfactory/centos.dockerfile.epp',{
         'puppetmaster' => $master_address,
       }),
-    require => File['/var/docker/centosagent/'],
-    notify => Docker::Image['centosagent'],
+    notify  => Docker::Image['centosagent'],
   }
 
   docker::image { 'centosagent':
     docker_dir => '/var/docker/centosagent/',
-    require     => File['/var/docker/centosagent/Dockerfile'],
```

Comment:
@samuelson or @binford2k Similarly, I think this is redundant because the Dockerfile has a `notify` that implies `before` this resource, but again, please verify at your convenience.